src/io/AtomicRelaxationReader.cc

```diff
@@ -0,0 +1,184 @@
+// Copyright 2020 UT-Battelle, LLC, and other Celeritas developers.
+// See the top-level COPYRIGHT file for details.
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file AtomicRelaxationReader.cc
+//---------------------------------------------------------------------------//
+#include "AtomicRelaxationReader.hh"
+
+#include <fstream>
+#include <sstream>
+#include "base/SoftEqual.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Construct the reader using the G4LEDATA environment variable to get the path
+ * to the data.
+ */
+AtomicRelaxationReader::AtomicRelaxationReader()
+{
+    const char* env_var = std::getenv("G4LEDATA");
+    CELER_VALIDATE(env_var, "Environment variable G4LEDATA is not defined.");
+    {
+        std::ostringstream os;
+        os << env_var << "/fluor";
+        fluor_path_ = os.str();
+    }
+    {
+        std::ostringstream os;
+        os << env_var << "/auger";
+        auger_path_ = os.str();
+    }
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Construct the reader with the path to the directory containing the data.
+ */
+AtomicRelaxationReader::AtomicRelaxationReader(const char* fluor_path,
+                                               const char* auger_path)
+    : fluor_path_(fluor_path), auger_path_(auger_path)
+{
+    CELER_EXPECT(!fluor_path_.empty());
+    CELER_EXPECT(!auger_path_.empty());
+
+    if (fluor_path_.back() == '/')
+        fluor_path_.pop_back();
+    if (auger_path_.back() == '/')
+        auger_path_.pop_back();
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Read the data for the given element.
+ */
+AtomicRelaxationReader::result_type
+AtomicRelaxationReader::operator()(int atomic_number) const
+{
+    CELER_EXPECT(atomic_number > 0 && atomic_number < 101);
+
+    // EADL does not provide transition data for Z < 6
+    result_type result;
+    if (atomic_number < 6)
+    {
+        return result;
+    }
+
+    std::string Z = std::to_string(atomic_number);
+
+    // Read fluorescence transition probabilities and subshell designators. All
+    // lines in the data file are 3 columns. The first line of each section is
+    // the subshell designator of the initial vacancy. Each of the following
+    // rows contains data for a different transition, where the columns are:
+    // designator of the shell of the electron that fills the vacancy,
+    // transition probability, and transition energy. A row of -1 marks the end
+    // of the shell data, and -2 marks the end of the file.
+    {
+        std::string   filename = fluor_path_ + "/fl-tr-pr-" + Z + ".dat";
+        std::ifstream infile(filename);
+        CELER_VALIDATE(infile, "Couldn't open '" << filename << "'");
+
+        int       des    = 0;
```

Comment:
Not gonna lie, every time I read this variable name I think of Karl Urban's character in Thor: Ragnarok 😂 
https://getyarn.io/yarn-clip/54bf355e-34fd-4d10-8e82-7a3564ebb1c7

Reply:
Still true a year and a half later 🤣